src/components/Table.tsx

```diff
@@ -0,0 +1,179 @@
+import get = require('lodash/get');
+import isPlainObject = require('lodash/isPlainObject');
+import map = require('lodash/map');
+import reverse = require('lodash/reverse');
+import sortBy = require('lodash/sortBy');
+import * as React from 'react';
+import FaSort = require('react-icons/lib/fa/sort');
+import styled from 'styled-components';
+
+import theme from '../theme';
+import Button from './Button';
+import { Flex } from './Grid';
+
+export interface TableDataRow {
+	[key: string]: any;
+}
+
+type TableSortFunction = <T>(a: T, b: T) => number;
+
+interface TableColumn {
+	field: string;
+	icon?: string;
+	label?: string | JSX.Element;
+	sortable?: boolean | TableSortFunction;
+	render?: (value: any) => string | JSX.Element;
+}
+
+interface TableProps {
+	columns: TableColumn[];
+	data: TableDataRow[];
+	// Optionally provide a key that should be used as a unique identifier for each row
+	rowKey?: string;
+}
+
+const BaseTable = styled.table`
+	width: 100%;
+	border-spacing: 0;
+
+	> thead {
+		background-color: #f2f2f2;
+
+		> tr > th {
+			padding-left: 40px;
+			padding-top: 10px;
+			padding-bottom: 10px;
+			font-size: 16px;
+		}
+	}
+
+	> tbody {
+		> tr {
+			> td {
+				font-size: 16px;
+				padding-top: 14px;
+				padding-bottom: 14px;
+				padding-left: 40px;
+			}
+
+			&:nth-of-type(even) {
+				background-color: #f8f8f8;
+			}
+		}
+	}
+`;
+
+const HeaderButton = styled(Button)`
+	display: block;
+`;
+
+/**
+ * Get the value specified by the `field` value
+ * If a `render` function is available, use it to get the display value.
+ */
+const renderField = (row: TableDataRow, column: TableColumn) => {
+	const value = get(row, column.field);
+
+	if (!value) {
+		return '';
+	}
+
+	return column.render ? column.render(value) : value;
```

Comment:
I think that when a render method exists, it should be fed with the raw value.
How about moving this before the `if (!value)` so that a render function can handle `undefined`/`null`/`0` in any crazy way?

Reply:
yep, I also think we should be passing in the full row to the render function as a second parameter, so that you can work with the presented data if necessary. For example, being able to access the devices uuid in the bulb component.

Reply:
Totally makes sense :+1: